File: Linked Lists/Remove Nth Node From End of List/solution2.cpp
// LeetCode - 19. Remove Nth Node From End of List
// Link - https://leetcode.com/problems/remove-nth-node-from-end-of-list/


/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode *start = new ListNode();
        start->next = head;
        ListNode *fast = start;
        ListNode *slow = start;
        int cnt = 1;
        
        while(cnt <= n){
            fast = fast->next;
            cnt++;
        }
        
        while(fast && fast->next != nullptr){
            slow = slow->next;
            fast = fast->next;
        }
        
        slow->next = slow->next->next;
        
        return start->next;
    }
   
};

File: Linked Lists/Palindrome Linked List/solution2.py
# LeetCode - 234. Palindrome Linked List
# Link - https://leetcode.com/problems/palindrome-linked-list/


# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        
        # Moving pointer till middle of LL
        slow = head
        fast = head
        temp = head
        
        while(fast.next and fast.next.next):
            slow = slow.next
            fast = fast.next.next
        
        if fast.next:
            slow = slow.next
        
        # Reversing LL from middle
        prev = ListNode()
        curr = slow
        nxt = ListNode()
        
        while(curr):
            nxt = curr.next
            curr.next = prev
            prev = curr
            curr = nxt
        
        slow = prev
        
        
        # Here comparing each value from start of LL and reversed LL from middle
        while(slow.next != None):
            if temp.val != slow.val:
                return False
            
            temp = temp.next
            slow = slow.next
        
        return True
        
